patches/tModLoader/Terraria/GameContent/ShopHelper.cs.patch

```diff
@@ -182,14 +182,35 @@
  			if (aShoppingBiome.IsInBiome(player)) {
  				AddHappinessReportText("HateBiome", new {
  					BiomeName = BiomeNameByKey(aShoppingBiome.NameKey)
-@@ -341,7 +_,10 @@
+@@ -341,9 +_,31 @@
  	private bool IsNotReallyTownNPC(NPC npc)
  	{
  		int type = npc.type;
 +		/*
  		if (type == 37 || type == 368 || type == 453)
 +		*/
-+		if (type == 37 || type == 368 || NPCID.Sets.ActsLikeTownNPC[type])
++		if (NPCID.Sets.IsNotReallyTownNPC[type])
  			return true;
  
  		return false;
++	}
++
++	internal void RegisterTownNPCMoodLocalizations(ModNPC npc)
++	{
++		if (npc.NPC.townNPC && !NPCID.Sets.IsTownPet[npc.NPC.type] && !Main.ShopHelper.IsNotReallyTownNPC(npc.NPC)) {
++
++			string prefix = $"Mods.{npc.Mod.Name}.TownNPCMood.{npc.Name}";
++			string[] keys = new string[] {
++				"Content", "NoHome", "FarFromHome", "LoveSpace", "DislikeCrowded", "HateCrowded", "LoveBiome", "LikeBiome", "DislikeBiome", "HateBiome", "LoveNPC", "LikeNPC", "DislikeNPC", "HateNPC", "LikeNPC_Princess"
++			};
++			foreach (var key in keys) {
++				string fullKey = $"{prefix}.{key}";
++				string defaultValueKey = "TownNPCMood." + key;
++				Language.GetOrRegister(fullKey, () => Language.GetTextValue(defaultValueKey));
```

Comment:
Should use a key reference to the default value, rather than `GetTextValue`

Reply:
I think this code is supposed to port pre-1.4.4 way of writing tooltips to 1.4.4 so the modder doesn't have to manually copypaste it

Reply:
My thought was the modder would see the entries in the .hjson and immediately be able to edit them to fit the voice and characterization of their town npc. If the modder just sees `"{$TownNPCMood.DislikeBiome}"`, they wouldn't know to use `{BiomeName}` in the text that they eventually write without looking at the game's source code.

Reply:
I agree, my misunderstanding. I'm not sure what this will do on non-english languages, but should be okay.